Sources/Account/Account+TransactionOperations.swift

```diff
@@ -281,5 +281,99 @@ extension Account {
                 }
             }
         }
+
+        func preparePresignedInputTransaction(
+            presignedInput: SignedContingentInput,
+            memoType: MemoType,
+            feeLevel: FeeLevel,
+            completion: @escaping (Result<PendingTransaction, TransactionPreparationError>)
+            -> Void
+        ) {
+            let amountToSend = presignedInput.requiredAmount
+            let amountToReceive = presignedInput.rewardAmount
+            let feeTokenId = amountToReceive.tokenId
+
+            metaFetcher.feeStrategy(for: feeLevel, tokenId: feeTokenId) {
+                switch $0 {
+                case .success(let feeStrategy):
+
+                    let (unspentTxOuts, ledgerBlockCount) =
+                        self.account.readSync {
+                            ($0.unspentTxOuts(tokenId: feeTokenId), $0.knowableBlockCount)
+                        }
+
+                    if ledgerBlockCount > presignedInput.tombstoneBlockIndex {
+                        serialQueue.async {
+                            completion(.failure(.invalidInput("Presigned Input Expired.")))
+                        }
+                        return
+                    }
+
+                    logger.info(
+                        "Preparing pre-signed input transaction with fee level...feeLevel: " +
+                            "\(feeLevel), unspentTxOutValues: " +
+                            "\(redacting: unspentTxOuts.map { $0.value })",
+                        logFunction: false)
+
+                    switch self.txOutSelector
```

Comment:
this statement is too long for a switch case, id put it on its own line and switch on its result, or remove the mapError part, similar to my suggestion on the previous file.

Reply:
will revisit nits after merge